### PR TITLE
Add support for LUKS encryption for OCP4

### PR DESCRIPTION
**Pull request by gauravpbankar**
This PR is mainly related to LUKS encryption which is required for PCI profile compliance.
To achieve this we have followed below document
https://docs.openshift.com/container-platform/4.11/installing/install_config/installing-customizing.html

This PR contains the variables and code changes that are required for encrypting block device in LUKS2 format.
This feature will enable Luks Encryption using Tang Servers on all master/worker nodes.
By default the encryption is disabled. User need to enable it in var.tfvars file during OCP deployment and also need to provide the passphrase and URL for each tang server.

For more details related to ocp4-playbook changes please refer below PR:
https://github.com/ocp-power-automation/ocp4-playbooks/pull/173

**Note: This is Day-1 activity and can be achieved by running ansible playbook tasks.**

Signed-off-by: Gaurav Bankar <Gaurav.Bankar@ibm.com>

### CODE REVIEW

**Comment by Rajalakshmi-Girish** [comment]
@gauravpbankar: PR is not mergeable.

<details>

The PR state is: BLOCKED

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by prb112** [comment]
@Rajalakshmi-Girish Hi 

Could you elaborate how we can move forward?

**Comment by Rajalakshmi-Girish** [comment]
> @gauravpbankar: PR is not mergeable.

You might need to get it approved by approvers in https://github.com/ocp-power-automation/ocp4-upi-powervs/blob/master/OWNERS
The UI of PR suggests 
image

**Comment by gauravpbankar** [comment]
@yussufsh @cs-zhang can you please review PR?

**Comment by yussufsh** [comment]
/hold
This PR need https://github.com/ocp-power-automation/ocp4-playbooks/pull/173 to be merged and used as reference for Terraform code.

File: modules/5_install/templates/install_vars.yaml
Comment (gauravpbankar) [inline modules/5_install/templates/install_vars.yaml:80]: Added required changes as suggested in below PR:
https://github.com/ocp-power-automation/ocp4-upi-powervs/pull/441

**Comment by gauravpbankar** [comment]
Hi @yu

> [ocp-power-automation/ocp4-playbooks#173](https://github.com/ocp-power-automation/ocp4-playbooks/pull/173)

Hi @yussufsh  as we can see above PR merged now can you please review powerVS PR now.

**Comment by yussufsh** [comment]
> Hi @yussufsh
> 
> > [ocp-power-automation/ocp4-playbooks#173](https://github.com/ocp-power-automation/ocp4-playbooks/pull/173)
> 
> Hi @yussufsh as we can see above PR merged now can you please review powerVS PR now.

You need to refer to the playbook's latest commit hash for this commit to work.

**Comment by gauravpbankar** [comment]
> > Hi @yussufsh
> > > [ocp-power-automation/ocp4-playbooks#173](https://github.com/ocp-power-automation/ocp4-playbooks/pull/173)
> > 
> > 
> > Hi @yussufsh as we can see above PR merged now can you please review powerVS PR now.
> 
> You need to refer to the playbook's latest commit hash for this commit to work.

Hi @yussufsh  

I have added required code change.

File: modules/5_install/variables.tf
Comment (yussufsh) [inline modules/5_install/variables.tf:118]:
```suggestion
variable "luks_compliant" { default = false }
```

File: modules/5_install/templates/install_vars.yaml
Comment (yussufsh) [inline modules/5_install/templates/install_vars.yaml:69]: `luks_options` is a list and not a string.

File: modules/5_install/templates/install_vars.yaml
Comment (yussufsh) [inline modules/5_install/templates/install_vars.yaml:79]: No need of this else block.

Comment (yussufsh) [reply modules/5_install/templates/install_vars.yaml:79]: By default luks.enable should be false.

File: var.tfvars
Comment (yussufsh) [inline var.tfvars:126]: Instead of mentioninig the values in comments, could you please add them to the values as they are any way defaults?
Eg:
```
#luks_filesystem_device = "/dev/mapper/root" #Set the Path of device to be luks encrypted
```

**Review by yussufsh — CHANGES_REQUESTED** [review]

File: modules/5_install/variables.tf
Comment (gauravpbankar) [reply modules/5_install/variables.tf]: Updated code as per your suggestion.

File: modules/5_install/templates/install_vars.yaml
Comment (gauravpbankar) [reply modules/5_install/templates/install_vars.yaml]: Updated code as per your suggestion.

File: modules/5_install/templates/install_vars.yaml
Comment (gauravpbankar) [reply modules/5_install/templates/install_vars.yaml]: Removed **else condition** from above code as per your suggestion as we are adding default value  luks.enabled to false in playbook as default value  mentioned below right?

ocp4-playbooks/playbooks/roles/ocp-config/defaults/main/main.yaml

```
luks:
   enabled: false
   filesystem_device: /dev/mapper/root
   format: xfs
   wipeFileSystem: true
   device: /dev/disk/by-partlabel/root
   label: luks-root
   options:
      - --cipher
      - aes-cbc-essiv:sha256
   wipeVolume: true
   name: root
```

File: var.tfvars
Comment (gauravpbankar) [reply var.tfvars]: Updated code as per mentioned above.

**Review by yussufsh — DISMISSED** [review]
/lgtm

**Comment by yussufsh** [comment]
/cc @Prajyot-Parab 
@gauravpbankar pls fix the fmt job errors.

**Comment by gauravpbankar** [comment]
> /cc @Prajyot-Parab @gauravpbankar pls fix the fmt job errors.

Fixed fmt job errors.

**Review by Prajyot-Parab — APPROVED** [review]
/approve

**Pull request merged**
